### PR TITLE
Fix loading the saved questions db for the first question

### LINKED ISSUE #24878: Newly created Model not an option when creating a new question
Probably related to #25144
 
tl;dr: We currently can't create a question from a freshly created model unless we reload the app.

Likewise when moving a model from one collection to another, the changes are not reflected in the model picker when creating a new question.

### To reproduce

1. Create a model
2. Reload the page
3. Create another model
2. Without refreshing the browser, create a question and choose Models as the source.

You will not see the newly created model.

If you reload the page and retry a new question, you will see the model.

### LINKED ISSUE #25144: First created question not in Saved Questions inside Data Selector until page reload
Probably related to #24878 

tl;dr: We currently can't create a question from a freshly created question (if the saved question is the first one) unless we reload the app.

### To reproduce

1. Have no questions or models saved.
2. Create and save a question
3. Do not reload the page
4. `+New` › `Question`

You will *not* see `Saved Questions` as an option in the Data Selector. 

If you reload the page and retry a new question, you will see Saved Questions as an option.

### PR DESCRIPTION
**Pull request by ranquild**
Fixes https://github.com/metabase/metabase/issues/25144
Fixes https://github.com/metabase/metabase/issues/24878
Reproduces https://github.com/metabase/metabase/issues/25144
Reproduces https://github.com/metabase/metabase/issues/24878

### CODE REVIEW

File: frontend/src/metabase/query_builder/actions/core/core.js [hunk 1]
```diff
@@ -231,11 +231,6 @@ export const apiUpdateQuestion = (question, { rerunQuery = false } = {}) => {
     // (some of the old alerts might be removed during update)
     await dispatch(fetchAlertsForQuestion(updatedQuestion.id()));
 
-    // remove the databases in the store that are used to populate the QB databases list.
-    // This is done when saving a Card because the newly saved card will be eligible for use as a source query
-    // so we want the databases list to be re-fetched next time we hit "New Question" so it shows up
-    dispatch(setRequestUnloaded(["entities", "databases"]));
```
Comment (ranquild) [inline frontend/src/metabase/query_builder/actions/core/core.js:237]: This line doesn't work because it needs to be `["entities", "databases_list"]` to unload lists.

File: frontend/src/metabase/query_builder/components/DataSelector/DataSelector.jsx [hunk 2]
```diff
@@ -212,13 +212,13 @@ const DataSelector = _.compose(
           entityQuery: ownProps.databaseQuery,
         }) ||
         [],
-      hasFetchedDatabasesWithTablesSaved: !!Databases.selectors.getList(state, {
+      hasLoadedDatabasesWithTablesSaved: Databases.selectors.getLoaded(state, {
```
Comment (ranquild) [inline frontend/src/metabase/query_builder/components/DataSelector/DataSelector.jsx:215]: Renamed `fetched` to `loaded` to match existing semantics.

File: frontend/src/metabase/query_builder/actions/core/core.js [hunk 3]
```diff
@@ -179,10 +179,10 @@ export const apiCreateQuestion = question => {
       .setResultsMetadata(resultsMetadata)
       .reduxCreate(dispatch);
 
-    // remove the databases in the store that are used to populate the QB databases list.
-    // This is done when saving a Card because the newly saved card will be eligible for use as a source query
-    // so we want the databases list to be re-fetched next time we hit "New Question" so it shows up
-    dispatch(setRequestUnloaded(["entities", "databases"]));
+    const databases = Databases.selectors.getList(getState());
+    if (databases && !databases.some(d => d.is_saved_questions)) {
```
Comment (ranquild) [inline frontend/src/metabase/query_builder/actions/core/core.js:183]: It isn't possible to write a reducer that updates not individual entities but list results, so let's do what the deleted line was supposed to do but conditionally to avoid extra requests.

**Review by gusaiani — APPROVED** [review]
Thanks very much, @ranquild.

Would still be interested to see how you went about debugging this one.
